agents/sorenson_dlm/dlm_agent.py

```diff
@@ -0,0 +1,349 @@
+        **Task** - Perform first time setup of the Sorenson DLM communication.
+
+        Parameters:
+            auto_acquire (bool, optional): Default is False. Starts data
+                acquisition after initialization if True.
+            acq_params (dict, optional): Params to pass to acq process if
+                auto_acquire is True.
+            force (bool, optional): Force initialization, even if already
+                initialized. Defaults to False.
+        """
+
+        if self.initialized and not params['force']:
+            self.log.info("DLM already initialized Returning...")
+            return True, "Already initialized"
+
+        with self._lock.acquire_timeout(timeout=3, job='init') as acquired:
+            if not acquired:
+                self.log.warn(f"Could not start init because "
+                              f"{self._lock.job} is already running")
+                return False, "Could not acquire lock"
+            session.set_status('running')
+            try:
+                self.dlm = DLM(self.ip_address, int(self.port))
+            except socket.timeout as e:
+                self.log.error("DLM power supply has timed out"
+                               + f"during connect with error {e}")
+                return False, "Timeout"
+
+        # Start data acquisition if requested
+        if params.get('auto_acquire', False):
+            self.agent.start('acq', params.get('acq_params', None))
+
+        self.initialized = True
+        return True, 'DLM module initialized.'
+
+    def _stop_acq(self, session, params=None):
+        """
+        End voltage data acquisition
+        """
+        if self.take_data:
+            self.take_data = False
+            return True, 'requested to stop taking data.'
+        else:
+            return False, 'acq is not currently running'
+
+
+def make_parser(parser=None):
+    """Build the argument parser for the Agent. Allows sphinx to automatically
+    build documentation based on this function.
+    """
+    if parser is None:
+        parser = argparse.ArgumentParser()
+
+    pgroup = parser.add_argument_group('Agent Options')
+    pgroup.add_argument('--ip-address', type=str, help="Serial-to-ethernet "
+                        + "converter ip address")
+    pgroup.add_argument('--port', type=int, help="Serial-to-ethernet "
+                        + "converter port")
+    pgroup.add_argument('--mode', default='acq',
+                        choices=['idel', 'acq'])
+    return parser
+
+
+if __name__ == '__main__':
+    parser = make_parser()
+    args = site_config.parse_args(agent_class='DLMAgent',
+                                  parser=parser)
+    # Automatically acquire data if request (default)
+    init_params = False
+    if args.mode == 'init':
+        init_params = {'auto_acquire': False,
+                       'acq_params': {},
+                       'configfile': args.configfile}
```

Comment:
`init` isn't a valid mode, per the modes listed under `choices` in the `--mode` argument of the parser. You can add that as a mode, but need to make sure the parameters here in this dictionary, which will be passed to `init_dlm` are valid as well.

Reply:
Removing this block, I don't think it's needed